cpp/src/fil/fil.cu

```diff
@@ -139,8 +169,8 @@ struct forest {
     num_trees_                       = params->num_trees;
     algo_                            = params->algo;
     output_                          = params->output;
-    threshold_                       = params->threshold;
-    global_bias_                     = params->global_bias;
+    threshold_                       = static_cast<real_t>(params->threshold);
```

Comment:
Not necessarily saying we should, but would it be possible to templatize `params` here such that the threshold is already of the correct type?

Reply:
I made those parameters `double` in `forest_params` and corresponding test parameters. This looks simpler than templatizing them, and they are not used in any performance-critical code.